### PR TITLE
Updates to get tests running with correct weak/strong checks

### PR DESCRIPTION
**Pull request by kevmoo**
...in bleeding edge SDK

### CODE REVIEW

File: pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart [hunk 1]
```diff
@@ -30,10 +30,16 @@ import 'package:test_api/src/suite_channel_manager.dart'; // ignore: implementat
...
-    RemoteListener.start(getMain,
-        hidePrints: hidePrints, beforeLoad: beforeLoad);
+    RemoteListener.start(
+      getMain,
+      hidePrints: hidePrints,
+      beforeLoad: beforeLoad,
+    )
+    // TODO: this cast should not be needed
+    // and yet it fails at SDK dc1d7c87205 without it
+    as StreamChannel<Object>;
```
Comment (kevmoo) [inline pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart:42]: This is a head scratcher, but absolutely necessary to get things working!

Comment (natebosch) [reply pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart:42]: What do you mean by "fails"? At runtime? Statically?

Comment (kevmoo) [reply pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart:42]:
```
  Failed to load "test/isolate_channel_test.dart":
  Unable to spawn isolate: ../test/pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart:35:20: Error: A value of type 'StreamChannel<dynamic>' can't be assigned to a variable of type 'StreamChannel<Object>'.
   - 'StreamChannel' is from 'package:stream_channel/stream_channel.dart' ('lib/stream_channel.dart').
   - 'Object' is from 'dart:core'.
      RemoteListener.start(getMain,
                     ^
```

Comment (kevmoo) [reply pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart:42]: statically!

Comment (natebosch) [reply pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart:42]: Is one of these libraries not opted in, and the other is?

Comment (kevmoo) [reply pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart:42]: I'm running w/ the bleeding SDK bits that won't let things be mixed – both are opted-in

**Comment by kevmoo** [comment]
@natebosch – LGTY?

**Comment by natebosch** [comment]
> LGTY?

No, I don't think we should be submitting workarounds until we understand them and have issues filed in the SDK.

**Comment by kevmoo** [comment]
> > LGTY?
> 
> No, I don't think we should be submitting workarounds until we understand them and have issues filed in the SDK.

I'll get going on the SDK issue.

File: pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart (hunk 1 above)
Comment (kevmoo) [reply pkgs/test_core/lib/src/runner/plugin/remote_platform_helpers.dart]: Turns out this **is not** needed. User error...

**Comment by kevmoo** [comment]
> > > LGTY?
> > 
> > 
> > No, I don't think we should be submitting workarounds until we understand them and have issues filed in the SDK.
> 
> I'll get going on the SDK issue.

Not sure what I did wrong, but it turns out the cast is NOT needed.

PTAL, @natebosch

**Comment by kevmoo** [comment]
@jakemac53 – this is much less scary now

**Pull request merged**